collector/client.go

```diff
@@ -207,6 +215,18 @@ func (c *CacheObjectClient) GetInfos() (types.Counters, error) {
 }
 
 func (ch *connectionHandlerImpl) connect() (net.Conn, error) {
+	if ch.certificate != "" && ch.key != "" {
+		cert, err := tls.LoadX509KeyPair(ch.certificate, ch.key)
+		if err != nil {
+			return nil, err
+		}
+		tlsconfig := tls.Config{Certificates: []tls.Certificate{cert}}
+		return tls.Dial("tcp", fmt.Sprintf("%s:%d", ch.hostname, ch.port), &tlsconfig)
```

Comment:
Ideally, you should use [net.JoinHostPort](https://pkg.go.dev/net#JoinHostPort) for concatenating hostname and port, since this `fmt.Sprintf("%s:%d", ...)` approach does not correctly handle literal IPv6 addresses (it won't add the necessary square brackets around the IPv6 address).

Reply:
Makes sense. I'll change it.

Reply:
Done